src/Client/SogeCommerceGateway.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+/*
+ * This file is part of akawaka/sylius-soge-commerce-plugin
+ *
+ * AKAWAKA
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Akawaka\SyliusSogeCommercePlugin\Client;
+
+use Akawaka\SyliusSogeCommercePlugin\Exception\FailedToCancelPaymentException;
+use Akawaka\SyliusSogeCommercePlugin\Exception\SogeCommerceApiNotActivatedException;
+use Akawaka\SyliusSogeCommercePlugin\Payum\PaymentGatewayFactory;
+use GuzzleHttp\ClientInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Sylius\Component\Payment\Model\PaymentInterface;
+use Webmozart\Assert\Assert;
+
+final class SogeCommerceGateway implements SogeCommerceGatewayInterface
+{
+    public function __construct(
+        private ClientInterface $client,
+        private OrderIdTransformerInterface $orderIdTransformer,
+    ) {
+    }
+
+    public function createFormToken(PaymentMethodInterface $method, OrderInterface $order): string
+    {
+        $gatewayConfig = $method->getGatewayConfig();
+        Assert::notNull($gatewayConfig);
+
+        if (PaymentGatewayFactory::FACTORY_NAME !== $gatewayConfig->getFactoryName()) {
+            throw new \LogicException(sprintf('Method with code "%s" is not a valid "%s" method.', $method->getCode() ?? '', PaymentGatewayFactory::FACTORY_NAME));
+        }
+
+        $config = $gatewayConfig->getConfig();
+        $user = $config['user'] ?? '';
+        Assert::string($user);
+        $password = $config['password'] ?? '';
+        Assert::string($password);
+
+        $authorization = base64_encode(sprintf('%s:%s', $user, $password));
+
+        $payment = $order->getPayments()->last();
+        Assert::isInstanceOf($payment, PaymentInterface::class);
+
+        $response = $this->client->request(
+            'POST',
+            'https://api-sogecommerce.societegenerale.eu/api-payment/V4/Charge/CreatePayment',
```

Comment:
I think a class/enum holding routes would be create

Reply:
I disagree. I believe the client's base URI should be configured so that only the action needs to be specified. I didn't do this because older versions of Sylius used Guzzle instead of Symfony's HTTP client.

If you think it should be moved to constants, I can still do that, but I don't think it would add much.